services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -0,0 +1,887 @@
+                 changePropertyValue(flow1.destination, "vlanId", flow2.source.vlanId),
+                 changePropertyValue(flow2.source, "vlanId", flow1.destination.vlanId),
+                 changePropertyValue(flow2.destination, "vlanId", flow1.source.vlanId)],
+
+                [changePropertyValue(flow1.source, "portNumber", flow2.destination.portNumber),
+                 changePropertyValue(flow1.destination, "portNumber", flow2.source.portNumber),
+                 changePropertyValue(flow2.source, "portNumber", flow1.destination.portNumber),
+                 changePropertyValue(flow2.destination, "portNumber", flow1.source.portNumber)],
+
+                [changePropertyValue(flow1.source, "datapath", flow2.destination.datapath),
+                 changePropertyValue(flow1.destination, "datapath", flow2.source.datapath),
+                 changePropertyValue(flow2.source, "datapath", flow1.destination.datapath),
+                 changePropertyValue(flow2.destination, "datapath", flow1.source.datapath)],
+        ]
+    }
+
+    @Unroll
+    def "Able to swap endpoints (#description) for two flows with different source and destination switches"() {
+        given: "Two flows with different source and destination switches"
+        flowHelper.addFlow(flow1)
+        flowHelper.addFlow(flow2)
+
+        when: "Try to swap endpoints for flows"
+        def response = northbound.swapFlowEndpoint(
+                new SwapFlowPayload(flow1.id, flow1Src, flow1Dst),
+                new SwapFlowPayload(flow2.id, flow2Src, flow2Dst))
+
+        then: "Endpoints are successfully swapped"
+        verifyEndpoints(response, flow1Src, flow1Dst, flow2Src, flow2Dst)
+        verifyEndpoints(flow1, flow2, flow1Src, flow1Dst, flow2Src, flow2Dst)
+
+        and: "Flows validation doesn't show any discrepancies"
+        validateFlows(flow1, flow2)
+
+        and: "Switch validation doesn't show any missing/excess rules and meters"
+        validateSwitches(flow1SwitchPair)
+        validateSwitches(flow2SwitchPair)
+
+        and: "Delete flows"
+        [flow1, flow2].each { flowHelper.deleteFlow(it.id) }
+
+        where:
+        description << ["src1 <-> src2", "dst1 <-> dst2", "src1 <-> dst2", "dst1 <-> src2"]
+        flow1SwitchPair << [getTopologyHelper().getNotNeighboringSwitchPair()] * 4
+        flow2SwitchPair << [getDifferentNotNeighboringSwitchPair(flow1SwitchPair)] * 4
+        flow1 << [getFirstFlow(flow1SwitchPair, flow2SwitchPair)] * 4
+        flow2 << [getSecondFlow(flow1SwitchPair, flow2SwitchPair, flow1)] * 4
+        [flow1Src, flow1Dst, flow2Src, flow2Dst] << [
+                [flow2.source, flow1.destination, flow1.source, flow2.destination],
+                [flow1.source, flow2.destination, flow2.source, flow1.destination],
+                [flow2.destination, flow1.destination, flow2.source, flow1.source],
+                [flow1.source, flow2.source, flow1.destination, flow2.destination]
+        ]
+    }
+
```

Comment:
I feel like all the above 8 tests can be combined into one, but with extended, bigger 'where' block. Or is that intentional to improve readability of 'where' blocks?